src/rt-app_parse_config.c

```diff
@@ -811,7 +811,7 @@ parse_thread_data(char *name, struct json_object *obj, int index,
 		data->phases = malloc(sizeof(phase_data_t) * data->nphases);
 		parse_thread_phase_data(obj,  &data->phases[0], opts, (long)data);
 		/* Get loop number */
-		data->loop = -1;
+		data->loop = get_int_value_from(obj, "loop", TRUE, -1);
 	}
```

Comment:
Right. So, if I revert Viresh' fix and use the following json, the thread only executes once and then terminates (while it should execute indefinetely)
```
{
        "tasks" : {
                "thread0" : {
                        "instance" : 1,
                        "cpus"     : [0],
                        "run"      : 1000000,
                        "sleep"    : 100000
                }
        },
        "global" : {
                "duration" : -1,
                "calibration" : 585,
                "default_policy" : "SCHED_OTHER",
                "pi_enabled" : false,
                "lock_pages" : false,
                "logdir" : "./",
                "log_size" : 10,
                "log_basename" : "rt-app"
        }
}
```
I guess that was the problem.
The final result of this series should fix both cases.